src/main/kotlin/graphics/scenery/backends/vulkan/VulkanTexture.kt

```diff
@@ -327,13 +329,24 @@ open class VulkanTexture(val device: VulkanDevice,
                 }
 
                 val storage = memAlloc(data.remaining() / 3 * 4)
-                val view = data.duplicate()
-                val tmp = ByteArray(pixelByteSize * 3)
-                val alpha = (0 until pixelByteSize).map { 255.toByte() }.toByteArray()
+                val view = data.duplicate().order(ByteOrder.LITTLE_ENDIAN)
+                val tmp = ByteArray(channelBytes * 3)
+                val alpha = when(gt.type) {
+                    is UnsignedByteType -> (0 until channelBytes).map { 255.toByte() }.toByteArray()
```

Comment:
I know this is my doing, but it'd be great if you could add the correct byte sequence for each type here.